sparse/core.py

```diff
@@ -641,11 +656,14 @@ def elemwise(self, func, *args, **kwargs):
                    sorted=self.sorted)
 
     def elemwise_binary(self, func, other, *args, **kwargs):
-        assert isinstance(other, COO)
+        if not isinstance(other, COO):
+            raise ValueError("Performing this operation would produce "
+                             "a dense result: %s" % str(func))
```

Comment:
Hrm, this isn't always true though.  For example `x > 5` or `x + 0` (which comes up when using the `sum` function) are both common case situations and should both work efficiently.  Is there some way that we can test against zero here?

Reply:
It's already being checked in `_perform_op`. That one handles operations with scalars pretty nicely, and is general. I've also added a test for it (`test_elemwise_scalar`)

Reply:
Though I can move it in here and refactor if you like.